Dockerfile

```diff
@@ -0,0 +1,155 @@
+ARG JUPYTER_MINIMAL_VERSION=lab-3.3.2@sha256:a4bf48221bfa864759e0f248affec3df1af0a68ee3e43dfc7435d84926ec92e8
+FROM jupyter/minimal-notebook:${JUPYTER_MINIMAL_VERSION} as base
+
+
+LABEL maintainer="iavarone"
+
+ENV JUPYTER_ENABLE_LAB="yes"
+# autentication is disabled for now
+ENV NOTEBOOK_TOKEN=""
+ENV NOTEBOOK_BASE_DIR="$HOME/work"
+
+USER root
+
+RUN apt-get update && \
+  apt-get install -y --no-install-recommends \
+  ffmpeg \
+  make \
+  dvipng \
+  gosu \
+  octave \
+  gnuplot \
+  liboctave-dev \
+  bc \
+  ghostscript \
+  texlive-xetex \
+  texlive-fonts-recommended \
+  texlive-latex-recommended \
+  texlive-fonts-extra \
+  zip \
+  fonts-freefont-otf \
+  && \
+  apt-get clean && rm -rf /var/lib/apt/lists/*   
+
+RUN pip --no-cache --quiet install --upgrade \
+  pip \
+  setuptools \
+  wheel
+
+# Service (dakota) specific installation
+# --------------------------------------------------------------------
+
+WORKDIR /opt
+
+# Dependencies for compilation
+RUN apt-get update && \
+  apt-get install -y --no-install-recommends \
+  gcc \
+  cmake \
+  libboost-all-dev \
+  libblas-dev \
+  liblapack-dev \
+  libopenmpi-dev \
+  openmpi-bin \
+  gsl-bin \
+  libgsl-dev \
+  perl \
+  libhdf5-dev \
+  gfortran
+
+# Download dakota tar
+
+ENV INSTALL_DIR /opt/dakota
+
+RUN wget https://dakota.sandia.gov/sites/default/files/distributions/public/dakota-6.16.0-public-src-cli.tar.gz && \
+  tar -xzvf dakota-6.16.0-public-src-cli.tar.gz && \
+  rm -rf dakota-6.16.0-public-src-cli.tar.gz
+
+RUN cmake -D CMAKE_INSTALL_PREFIX=${INSTALL_DIR} \
+  -D CMAKE_C_FLAGS="-O2" -D CMAKE_CXX_FLAGS="-O2"  \
+  -D CMAKE_Fortran_FLAGS="-O2" \ 
+  -D DAKOTA_HAVE_GSL:BOOL=TRUE \ 
+  -D HAVE_QUESO:BOOL=TRUE \ 
+  -D DAKOTA_HAVE_MPI:BOOL=TRUE \ 
+  -D DAKOTA_HDF5:BOOL=TRUE \
+  -D Boost_NO_BOOST_CMAKE:BOOL=TRUE \
+  dakota-6.16.0-public-src-cli
+
+# compile and install
+RUN make -j 4
+RUN make install
+RUN rm -r dakota-6.16.0-public-src-cli
+
+
+ENV PATH="${PATH}:$INSTALL_DIR/bin:$INSTALL_DIR/share/dakota/test:$INSTALL_DIR/gui"
+RUN echo "export PATH=$INSTALL_DIR/bin:$INSTALL_DIR/share/dakota/test:$INSTALL_DIR/gui:${PATH}" >> ~/.bashrc
+
+WORKDIR ${HOME}/test_dakota
+
+RUN cp ${INSTALL_DIR}/share/dakota/examples/users/rosen_multidim.in ${HOME}/test_dakota
+
+# test Dakota
+RUN dakota -v
```

Comment:
I have two suggestions here:
- try to use env var with for the version version (eg DAKOTA_VERSION="6.16.0"), this way you can bump it more easily in the future.
- add all this in one unique command, like you se when installing python dependencies below. This allows for less docker layers and smaller images.

TIP: to achieve above suggestions you need to put all the `ENV` declarations at the beginning, also the one altering the PATH it will still work event if the directories have not been created

Reply:
Thanks for the tips. I implemented them!